src/Juvix/FrontendContextualise/ModuleOpen/Environment.hs

```diff
@@ -100,3 +159,189 @@ lookupModMap s =
 removeModMap ::
   HasState "modMap" ModuleMap m => Symbol -> m ()
 removeModMap s = Juvix.Library.modify @"modMap" (Map.delete s)
+
+--------------------------------------------------------------------------------
+-- fully resolve module opens
+--------------------------------------------------------------------------------
+
+-- Precondition ∷ old and new context must be in the same context
+-- also terms must be in either the old map or the new map
+-- any place where this is inconsistent will break resolution
+
+-- | @populateModMap@ populates the modMap with all the global opens
+-- in the module
+populateModMap ::
+  WorkingMaps m => m ()
+populateModMap = do
+  old <- get @"old"
+  new <- get @"new"
+  open <- Juvix.Library.ask @"openMap"
+  modM <- get @"modMap"
+  let curr = pure Context.topLevelName <> Context.currentName old
+  case open Map.!? curr of
+    Nothing ->
+      pure ()
+    Just opens ->
+      -- TODO ∷
+      -- explicts and implicits for now work the same
+      put @"modMap" (foldr f modM opens)
```

Comment:
you can just use `modify` instead of `get` and `put`

Reply:
I'd need to flip the foldr then

Reply:
still I think it's nicer, nbd though